svm/src/transaction_processor.rs

```diff
@@ -361,9 +361,9 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
                                 if let Some(index) =
                                     callbacks.account_matches_owners(key, program_owners)
                                 {
-                                    program_owners
-                                        .get(index)
-                                        .map(|owner| entry.insert((owner, 1)));
+                                    if let Some(owner) = program_owners.get(index) {
+                                        entry.insert((owner, 1));
+                                    }
```

Comment:
Although the `if` case is not explicitly detected in clippy, I think using map is preferred: https://rust-lang.github.io/rust-clippy/master/index.html#/manual_map.

Reply:
I see. I actually found using map in this case confusing, because the only purpose of it is to do something with the slice element when the index is within the slice boundaries, and the result of map is discarded. Wouldn't say it's more concise either...